app/blueprints/search.py

```diff
@@ -1,11 +1,42 @@
+from functools import wraps
+
 from flask import Blueprint, current_app, jsonify, request
-from index_search import search as azure_index_search
+from index_search import search
+
+from app.finesse_data import fetch_data
+
+search_blueprint = Blueprint("finesse", __name__)
+
 
-search = Blueprint("finesse", __name__)
+def require_non_empty_query(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        query = request.json.get("query")
+        if not query:
+            return jsonify({"message": "Search query cannot be empty"}), 400
+        return f(*args, **kwargs)
+
+    return decorated_function
+
+
+@search_blueprint.route("/azure", methods=["POST"])
+@require_non_empty_query
+def search_azure():
+    query = request.json["query"]
+    try:
+        results = search(query, current_app.config["AZURE_CONFIG"])
+        return jsonify(results)
+    except Exception as e:
+        return jsonify({"error": str(e)}), 500
```

Comment:
CodeQL seems to have a good point. Probably want to only have the stacktrace turned on when FINESSE_BACKEND_DEBUG_MODE=true

Reply:
Flask indeed has a mechanism to automatically handle this based on the debug mode. We could just not handle the exceptions and let flask do it but this will fail the tests.